servers/src/grin/server.rs

```diff
@@ -506,13 +506,17 @@ impl Server {
 			total_difficulty: head.total_difficulty(),
 		};
 
-		let header_tip = self.chain.header_head()?;
-		let header = self.chain.get_block_header(&header_tip.hash())?;
-		let header_stats = ChainStats {
-			latest_timestamp: header.timestamp,
-			height: header.height,
-			last_block_h: header.prev_hash,
-			total_difficulty: header.total_difficulty(),
+		let header_tip = self.chain.try_header_head()?;
+		let header_stats = if let Some(tip) = header_tip {
+			let header = self.chain.get_block_header(&tip.hash())?;
```

Comment:
Nitpick - perhaps `match` would be more readable here, `if let` shines when we care about one variant

Reply:
You're right that is better :)